src/gui/Src/Gui/StructWidget.cpp

```diff
@@ -109,44 +110,61 @@ void StructWidget::typeAddNode(void* parent, const TYPEDESCRIPTOR* type)
         dtype.type.addr = type->addr;
         dtype.type.offset = type->offset;
         dtype.type.id = type->id;
-        dtype.type.sizeBits = type->sizeBits * 8;
+        dtype.type.sizeBits = type->sizeBits;
         dtype.type.callback = type->callback;
         dtype.type.userdata = type->userdata;
+        dtype.type.typeName = type->typeName;
         dtype.type.bitOffset = 0;
     }
+
     dtype.name = highlightTypeName(dtype.type.name);
+    dtype.typeName = QString(dtype.type.typeName);
     dtype.type.name = nullptr;
+    dtype.type.typeName = nullptr;
+
     QStringList text;
     auto columnCount = ui->treeWidget->columnCount();
     for(int i = 0; i < columnCount; i++)
         text.append(QString());
 
     text[ColOffset] = "+0x" + ToHexString(dtype.type.offset);
     text[ColField] = dtype.name;
-    if(dtype.type.offset == 0 && true)
+    if(dtype.type.offset == 0)
         text[ColAddress] = QString("<u>%1</u>").arg(ToPtrString(dtype.type.addr + dtype.type.offset));
     else
         text[ColAddress] = ToPtrString(dtype.type.addr + dtype.type.offset);
     text[ColSize] = "0x" + ToHexString(dtype.type.sizeBits / 8);
     text[ColValue] = ""; // NOTE: filled in later
+
     QTreeWidgetItem* item = nullptr;
     if(parent == nullptr)
     {
-        item = new QTreeWidgetItem(ui->treeWidget, text);
+        if(mInsertIndex != -1)
+        {
+            item = new QTreeWidgetItem(text);
+            ui->treeWidget->insertTopLevelItem(mInsertIndex, item);

```

Comment:
Here it should do `mInsertIndex = -1` so the next type won't be inserted in a 'random' position.